### DISCUSSION TITLE
Statistical Analysis

### DISCUSSION THREAD

**Discussion by Gyuknj** [pmlmodelling/nctoolkit#49] [Q&A]
I need help with codes to run these statistical analysis using nctoolkit: 
1. Regression,
2. Calculate the lagged correlation
3. Calculate the cross-correlation functions
4. Calculate the growth rate and also the explained variances.

**Comment by robertjwilson** [comment]
You have asked 5 questions. Have you looked through the nctoolkit documentation to see if there are relevant methods?

**Reply by Gyuknj** [reply]
OH. Sorry for asking too many questions at a time. I have looked at the nctoolkit documentation and these analysis I mention in my questions are not there.

**Reply by robertjwilson** [reply]
The documentation is extensive. If there is something that is missing that you would like and have ideas for how to add it, please raise an issue. I can then look into it.

**Reply by Gyuknj** [reply]
Yes it is! I have gone through the documentation (about 127 pages) but can't find regression or any of the analysis mentioned. Kindly, assist me with the command line or link where I can get it and others if possible. Thank you!

**Reply by robertjwilson** [reply]
The documentation lists all available methods. nctoolkit is not a package for statistical analysis, and does not claim to be. With all due respect, if you ask people questions try to be more considerate. I took the time to solve your problem with CDO 2.0.4, and then you immediately threw a bunch of questions at me without providing any evidence you had to tried to solve the problem. 

As I said, if there is functionality missing in nctoolkit that you would like and have some idea for how to add it, please feel free to raise an issue.

**Reply by Gyuknj** [reply]
Okay, I understand! Thank you for helping me out to solve the problem with CDO 2.0.4. 

Also, sorry if I am not considerate in my questions. Maybe its because I am liking nctoolkit already and prefer to use for my analysis. 
I have tried alot of options but I was not giving the results I wanted. I did not share because it is bulky. 

I compared the results I got from the correlation analysis with the previous results, the results I got here is better and that was why I asked if I could do regression and other analysis usin the tool. 

Thank you for your time and assistance.

**Reply by robertjwilson** [reply]
I am happy to help with any questions. But please take the time to ask questions that are totally clear and ask them separately.  For example, "Calculate the growth rate" could mean all kinds of things in marine ecology.

**Reply by Gyuknj** [reply]
Alright! Sorry for stressing you this much with alot of question with little or no clearity. 

I have two datasets, namely; Sea surface temperature and 10m wind that span 1959 to 2022.
I used the script below written by you to compute the spatial correlation and it gave me good results.
 
import nctoolkit as nc
files = ["data/CMIP6_UKESM1-0-LL_Lmon_piControl_r1i1p1f2_gpp_1960-3059.nc",
"data/CMIP6_UKESM1-0-LL_Amon_piControl_r1i1p1f2_tas_1960-3059.nc"]
ds = nc.open_data(files)
ds.merge()
ds.cor_time(var1 = "gpp", var2 =  "tas")
ds.plot()

I have been trying to compute also the spatial regression for the two datasets but no success.
Kindly, assist me with a similar script for the spatial regression if it is possible. 

Thank you once again for your time and patience.

**Reply by robertjwilson** [reply]
As I said before, if you have separate questions please post them separately. I want discussions/issues to be of use to anyone using the website
